src/server/controllers/VerificationController.ts

```diff
@@ -84,16 +135,220 @@ export default class VerificationController extends BaseController implements IC
+        for (const id in contractWrappers) {
+            const contractWrapper = contractWrappers[id];
+            if (!isVerifiable(contractWrapper)) {
+                continue;
+            }
+            const inputData: InputData = { addresses: [contractWrapper.address], chain: contractWrapper.networkId, contract: contractWrapper.contract };
+
+            // TODO check if address is already verified? be careful not to permanently set the status
+            const matchPromise = this.verificationService.inject(inputData, config.localchain.url);
+            const match: Match = await matchPromise.catch((error: Error) => {
+                return {
+                    status: null,
+                    address: null,
+                    message: error.message
+                };
+            });
+
+            contractWrapper.status = match.status || "error";
+            contractWrapper.statusMessage = match.message;
+        }
+    }
+
+    private extractFiles(req: Request): PathBuffer[] {
+        let pathBuffers: PathBuffer[];
+        if (req.is('multipart/form-data')) {
+            if (!req.files) {
+                return null; // user only wants to check if the address has been verified
+            }
+            if (!req.files.files) {
+                throw new BadRequestError('The uploaded files should be under the "files" property');
+            }
+            const fileArr: fileUpload.UploadedFile[] = [].concat(req.files.files); // ensure an array, regardless of how many files received
+            pathBuffers = fileArr.map(f => ({ path: f.name, buffer: f.data }));
+
+        } else if (req.is('application/json')) {
+            if (!req.body.files) {
+                throw new BadRequestError('The uploaded files should be under the "files" property');
+            }
+            pathBuffers = [];
+            for (const name in req.body.files) {
+                const file = req.body.files[name];
+                const buffer = (Buffer.isBuffer(file) ? file : Buffer.from(file));
+                pathBuffers.push({ path: name, buffer });
+            }
+
+        } else {
+            throw new BadRequestError("Cannot perform the request for the provided Content-Type");
+        }
+
+        return pathBuffers;
+    }
+
+    private saveFiles(pathContents: PathContent[], session: MySession) {
+        if (!session.inputFiles) {
+            session.inputFiles = {};
+        }
+        
+        let inputSize = 0; // shall contain old buffer size + new files size
+        for (const id in session.inputFiles) {
+            const pc = session.inputFiles[id];
+            inputSize += pc.content.length;
+        }
+
+        pathContents.forEach(pc => inputSize += pc.content.length);
+
+        if (inputSize > VerificationController.MAX_INPUT_SIZE) {
+            const msg = "Too much session memory used. Delete some files or restart the session.";
+            throw new BadRequestError(msg); // TODO 413 Payload Too Large
```

Comment:
This return message should be added

Reply:
What do you mean? A PayloadTooLarge error class?

Reply:
Maybe PayloadTooLargeError can extend BadRequestError? Since that is bad request but it would be nice to return different error code for PayloadTooLargeError. It improves the API when used programmatically because it is easier to just check the status code.